Framework/Algorithms/src/ConvertUnits.cpp

```diff
@@ -412,10 +412,16 @@ bool ConvertUnits::getDetectorValues(const API::SpectrumInfo &spectrumInfo,
 
   if (!spectrumInfo.isMonitor(wsIndex)) {
     // The scattering angle for this detector (in radians).
-    if (signedTheta)
-      twoTheta = spectrumInfo.signedTwoTheta(wsIndex);
-    else
-      twoTheta = spectrumInfo.twoTheta(wsIndex);
+    try {
+      if (signedTheta)
+        twoTheta = spectrumInfo.signedTwoTheta(wsIndex);
+      else
+        twoTheta = spectrumInfo.twoTheta(wsIndex);
+    } catch (const std::runtime_error &e) {
+      Kernel::Logger g_log("ConvertUnits");
```

Comment:
Thanks for changing this. This way it has no effect on performance when `twoTheta` doesn't throw.

Just one minor cleanup: Creating a `Logger` is quite expensive and the Algorithm base class already creates one for each algorithm that is run so you should just be able to remove `Kernel::Logger g_log("ConvertUnits");` and `g_log.warning` will use the one from the base class.